programs/nft_api/src/nft_state.rs

```diff
@@ -0,0 +1,17 @@
+use bincode::serialize_into;
+use serde_derive::{Deserialize, Serialize};
+use solana_sdk::{instruction::InstructionError, pubkey::Pubkey, short_vec};
+
+#[derive(Serialize, Deserialize, Debug, Clone, Default, PartialEq)]
+pub struct NftState {
+    pub issuer_pubkey: Pubkey,
```

Comment:
Do we need the issuer_pubkey in the state? Seems like a program developer might choose to have only one issuing keypair so this information might just be redundant. Maybe useful as an audit trail?

Reply:
Yes, but only if we're using it to "namespace" the `id` field below. Say, for example, you wrote a separate program that'll transfer me 42 lamports as soon as `id = 123, issuer_pubkey = CRYPTOKITTIES_PUBKEY` was in possession of `owner_pubkey = YOU_PUBKEY`.  Can't go sending those lamports when some other issuer created an NFT with the same `id`.